clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -1179,6 +1195,113 @@ void StreamChecker::evalUngetc(const FnDescription *Desc, const CallEvent &Call,
   C.addTransition(StateFailed);
 }
 
+ProgramStateRef StreamChecker::ensureGetdelimBufferAndSizeCorrect(
+    SVal LinePtrPtrSVal, SVal SizePtrSVal, const Expr *LinePtrPtrExpr,
+    const Expr *SizePtrExpr, CheckerContext &C, ProgramStateRef State) const {
+  static constexpr char SizeGreaterThanBufferSize[] =
```

Comment:
Now it uses `llvm:StringLiteral`, but I think it (relatively) is. Most existing uses of `constexpr llvm::StringLiteral` in llvm are with `static`. And, IIUC, without `static`, the variable is going to live on the stack and be created each time (even though its value was already computed at compile time).

Probably a moot point, though, since this is not part of a hotpath. For consistency with other uses, I'd rather keep it, though.